test/python/transpiler/test_commutative_cancellation.py

```diff
@@ -76,7 +77,7 @@ def test_all_gates(self):
         expected = QuantumCircuit(qr)
         expected.append(RZGate(2.0), [qr[0]])
         expected.rx(1.0, qr[0])
-
+        expected.global_phase = 0.5
```

Comment:
I might have missed this in the PR summary message, but was this caused by the change in logic you made around the global phase handling?

Reply:
Yes. The previous code would only consider the last replaced gate when setting the new `global_phase`. I think this was essentially a typo of 
```
global_phase = new_phase
```
instead of 
```
global_phase += new_phase
```